### ISSUE TITLE
[ranges] Should `constexpr default_sentinel_t end()` be `const`-qualified?

### ISSUE THREAD

**Issue by hewillk** [cplusplus/draft#5507]
The `begin`/`end` of `basic_istream_view` is defined in [[range.istream.view]](https://eel.is/c++draft/range.istream.view)

```
template<movable Val, class CharT, class Traits = char_traits<CharT>>
class basic_istream_view : public view_interface<basic_istream_view<Val, CharT, Traits>> {
public:
  constexpr auto begin() {
    *stream_ >> value_;
    return iterator{*this};
  }
  constexpr default_sentinel_t end() const noexcept;
};
```
Even if `begin()` is non-`const`-qualified, since `end()` simply returns `default_sentinel`, it is still designed to be `const`-qualified member function, which allows us to call `as_const(r).end()` even if `as_const(r)` is not a `range`.

However, the `begin`/`end` of `chunk_view` for input ranges is defined in [[range.chunk.view.input]](https://eel.is/c++draft/range.chunk.view.input) as

```
template<view V>
class chunk_view : public view_interface<chunk_view<V>> {
public:
  constexpr outer-iterator begin();
  constexpr default_sentinel_t end() noexcept;
};
```
Even though `end()` can be `const`-qualified, it is still designed to be non-`const`-qualified, which makes `end()`'s `const`-qualification _consistent_ with `begin()`'s and prevents us from calling `as_const(r).end()`.

Do we need to resolve this inconsistency?

**Comment by jensmaurer** [comment]
This does not seem to be an issue that can be resolved editorially. Please file an LWG issue.

**Comment by jwakely** [comment]
FWIW I think we did just make that change during the review of `generator` but I agree it would be non-editorial to change anything already in the working draft.

**Comment by hewillk** [comment]
> FWIW I think we did just make that change during the review of `generator` but I agree it would be non-editorial to change anything already in the working draft.

Thanks for your information.
So did the `generator `finally drop the `const `of `end`? What were the decision of the committee and the consideration behind it? It would be great to know.

**Comment by jwakely** [comment]
Oh I got confused, `end()` was always `const` in the generator paper, but we added `const` to some other simple members that don't need to be non-const.

**Comment by frederick-vs-ja** [comment]
Resolved as LWG3710. Should be fixed by #5667 (d7a01d97fc63b692df1d4aa2f8b5266236daa5b7).
